src/ZipkinOpenTracing/Tracer.php

```diff
@@ -38,6 +37,59 @@ public function __construct(ZipkinTracing $tracing)
     {
         $this->tracer = $tracing->getTracer();
         $this->propagation = $tracing->getPropagation();
+        $this->scopeManager = new ScopeManager();
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function getScopeManager()
+    {
+        return $this->scopeManager;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function getActiveSpan()
+    {
+        $activeScope = $this->scopeManager->getActiveScope();
+        if ($activeScope === null) {
+            return null;
+        }
+
+        return $activeScope->getSpan();
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function startActiveSpan($operationName, $options = [])
+    {
+        if (!$options instanceof SpanOptions) {
+            $options = SpanOptions::create($options);
+        }
+
+        $child = $this->findChildInReferences($options->getReferences());
+        if ($child === null && $this->getActiveSpan() !== null) {
+            $child = $this->getActiveSpan()->getContext();
+        }
+
+        $options = SpanOptions::create(array_merge([
+            'tags' => $options->getTags(),
+            'start_time' => $options->getStartTime(),
+            'close_span_on_finish' => $options->getCloseSpanOnFinish(),
+        ], ($child ? ['child_of' => $child] : [])));
+
+        $span = $this->startSpan($operationName, $options);
+        $scope = $this->scopeManager->activate($span);
+
+        if ($span instanceof ZipkinOpenTracingSpan) {
```

Comment:
If you pass `$scope` and `$options->getCloseSpanOnFinish()` in the constructor then you don't need to do this.

Reply:
A span is constructed via `startSpan` call. On that call we are no aware of what the scope is (I assume startSpan does not activate the scope).